src/platform/packages/shared/kbn-unified-metrics-grid/src/components/metrics_grid_wrapper.tsx

```diff
@@ -0,0 +1,209 @@
+  const fullScreenZIndex = Number(euiTheme.levels.header) - 1;
+  return {
+    [METRICS_GRID_FULL_SCREEN_CLASS]: css`
+      z-index: ${fullScreenZIndex} !important;
+      position: fixed;
+      inset: 0;
+      background-color: ${euiTheme.colors.backgroundBasePlain};
+    `,
+    [METRICS_GRID_RESTRICT_BODY_CLASS]: css`
+      ${logicalCSS('height', '100vh')}
+      overflow: hidden;
+
+      .euiHeader[data-fixed-header] {
+        z-index: ${fullScreenZIndex - 1} !important;
+      }
+
+      .euiOverlayMask[data-relative-to-header='below'] {
+        ${logicalCSS('top', '0')}
+      }
+
+      .euiFlyout {
+        ${logicalCSS('top', '0')}
+        ${logicalCSS('height', '100%')}
+      }
+    `,
+  };
+};
+
+export const METRICS_GRID_WRAPPER_FULL_SCREEN_CLASS = 'metricsExperienceGridWrapper--fullScreen';
+export const METRICS_GRID_FULL_SCREEN_CLASS = 'metricsExperienceGrid--fullScreen';
+export const METRICS_GRID_RESTRICT_BODY_CLASS = 'metricsExperienceGrid--restrictBody';
+// Ensure full screen metrics grids are not covered by elements with a z-index
+const fullScreenBodyStyles = css`
+  *:not(
+      .euiFlyout,
+      .${METRICS_GRID_FULL_SCREEN_CLASS}, .${METRICS_GRID_FULL_SCREEN_CLASS} *,
+      [data-euiportal='true'],
+      [data-euiportal='true'] *
+    ) {
+    z-index: unset !important;
+  }
+`;
+
+const bodyClassesToToggle = [fullScreenBodyStyles, METRICS_GRID_WRAPPER_FULL_SCREEN_CLASS];
+const toggleMetricsGridFullScreen = (metricsGrid: HTMLElement) => {
+  const hasFullScreenClass = metricsGrid.classList.contains(METRICS_GRID_FULL_SCREEN_CLASS);
+
+  if (hasFullScreenClass) {
+    document.body.classList.add(...bodyClassesToToggle);
+  } else {
+    document.body.classList.remove(...bodyClassesToToggle);
+  }
+};
+
+export const useMetricsGridFullScreen = () => {
+  const metricsGridId = useGeneratedHtmlId({ prefix: 'metricsExperienceGrid' });
+  const [metricsGridWrapper, setMetricsGridWrapper] = useState<HTMLElement | null>(null);
+  const [metricsGrid, setMetricsGrid] = useState<HTMLElement | null>(null);
+
+  const findMetricsGridElement = useCallback<MutationCallback>(
+    (_, observer) => {
+      const foundMetricsGrid = document.getElementById(metricsGridId);
+
+      if (foundMetricsGrid) {
+        setMetricsGrid(foundMetricsGrid);
+        observer.disconnect();
+      }
+    },
+    [metricsGridId]
+  );
+
+  const handleMetricsGridFullScreenToggle = useCallback<MutationCallback>(() => {
+    if (metricsGrid) {
+      toggleMetricsGridFullScreen(metricsGrid);
+    }
+  }, [metricsGrid]);
+
+  useMutationObserver(metricsGridWrapper, findMetricsGridElement, {
+    childList: true,
+    subtree: true,
+  });
+
+  useMutationObserver(metricsGrid, handleMetricsGridFullScreenToggle, {
+    attributes: true,
+    attributeFilter: ['class'],
+  });
+
+  return { metricsGridId, metricsGridWrapper, setMetricsGridWrapper };
+};
+
+export interface MetricsGridWrapperProps
```

Comment:
Perhaps we could move everything above this line into a separate hook, wdyt? Also, does it make sense to add tests for `useMetricsGridFullScreen`?

Reply:
Regarding the tests, I checked if the classes are toggled correctly and if we have the correct number of calls and params.